neoteknic/Phue
# feat: modernize, add strong types, add phpstan, add php-cs-fixer, get…

… phpunit running
convert phpunit config-files to v. 9.5
add github actions
use promoted property constructors
remove useless comments and annotations
add phpstan-annotation
add TODO Entries for ideas to improve further
move config-files to the default location
fix namespaces in tests
improve tests
adjust readme


Hi. I know it is a big PR. Sorry for that.
This PR targets a new major version as there are BC brakes with the Strong types.


TODOs
- Test in real life application
- remove Temp added getters (private attributes testing is not available anymore in phpunit 9). 
   Needs to be refactored to check with reflections.
- Check if IDs int vs string (Existing Unittests say it is all int)
- Fix broken Tests:
3 Tests are failing. This is for changes with the new parameter for brigthness. I have no knowledge about that feature. Please fix that by yourself (as you introduced this parameter).
https://github.com/Chris53897/Phue/runs/7825345192?check_suite_focus=true

Please let me know what you think.
Greetz
